src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -377,77 +343,116 @@ using namespace std;
+        m_writer.block_end();
+
+        m_writer << "extern \"C\" const size_t get_debug_timer_microseconds(size_t index)\n";
+        m_writer.block_begin();
+        m_writer << "return (index < " << names.size()
+                 << " ? timers[index].get_total_microseconds() : 0);\n";
+        m_writer.block_end();
+
+        m_writer << "extern \"C\" const size_t get_debug_timer_call_count(size_t index)\n";
+        m_writer.block_begin();
+        m_writer << "return (index < " << names.size()
+                 << " ? timers[index].get_call_count() : 0);\n";
+        m_writer.block_end();
+        m_writer << "\n";
     }
-    writer << "// Declare all constants\n";
-    for (shared_ptr<Function> current_function : pass_manager.get_state().get_functions())
+}
+
+void runtime::gpu::GPU_ExternalFunction::emit_declare_constants()
+{
+    m_writer << "// Declare all constants\n";
+    for (shared_ptr<Function> current_function : m_pass_manager.get_state().get_functions())
     {
-        for (shared_ptr<Node> node : current_function->get_ordered_ops())
+        for (shared_ptr<Node> node : m_function_ordered_ops.at(current_function))
         {
             const op::Constant* c = dynamic_cast<ngraph::op::Constant*>(node.get());
             if (c)
             {
-                shared_ptr<descriptor::TensorView> tv = node->get_outputs()[0].get_tensor_view();
-                auto c_value_strings = c->get_value_strings();
-                writer << "static " << tv->get_tensor().get_element_type().c_type_string() << " *"
-                       << tv->get_tensor().get_name() << ";\n";
-                writer << "static " << tv->get_tensor().get_element_type().c_type_string() << " "
-                       << tv->get_tensor().get_name() << "_cpu[" << c_value_strings.size()
-                       << "] =\n";
-                writer << "{\n";
-                writer.indent++;
-                writer << emit_string_array(c_value_strings, 100 - writer.indent * 4);
-                writer.indent--;
-                writer << "\n};\n\n";
+                shared_ptr<descriptor::TensorView> tv =
+                    node->get_outputs()[0].get_tensor_view();
+                // get an allocator for transient per kernel gpu memory
+                GPUAllocator allocator = this->m_primitive_emitter->get_memory_allocator();
+                size_t idx = allocator.reserve_argspace(
+                    c->get_data_ptr(),
+                    tv->get_tensor().size() * tv->get_tensor().get_element_type().size());
+                m_writer << "static size_t " << tv->get_tensor().get_name() << "_idx = " << idx
+                         << ";\n";
+                m_writer << "static " << tv->get_tensor().get_element_type().c_type_string() << "* "
+                         << tv->get_tensor().get_name() << " = nullptr;\n";
                 m_variable_name_map[tv->get_tensor().get_name()] = tv->get_tensor().get_name();
             }
         }
     }
 
```

Comment:
I'm a little bit worried about these `static` declarations in the JIT. I have found in several instances that there were errors in application teardown post return in `main` when memory was statically declared in the JIT code. Is it necessary for these to be static? I'm guessing so, in which case it would be useful to test this in a production-like environment to understand if there are any side effects.

Reply:
*The way it works now need to be static.
*There is a way to change it but will need to get buffer address each time, which I think should be fine. 
*I don't understand why static variable should not be a problem in application teardown if use correctly.